controllers/controller_common.go

```diff
@@ -0,0 +1,29 @@
+// Copyright © 2019 Banzai Cloud
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package controllers
+
+import (
+	"github.com/go-logr/logr"
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+func logErrorAndRequeue(logger logr.Logger, msg string, err error) (ctrl.Result, error) {
+	logger.Error(err, msg)
+	return ctrl.Result{Requeue: true}, nil
```

Comment:
I think sadly this is not that easy. Returning `ctrl.Result{}, err` tells the controller to requeue  the request and set an exponential backoff strategy, on the other hand if we use `ctrl.Result{Requeue: true}, nil` it just tells the controller to requeue immediately. We need to distinguish these two. 
We should use `ctrl.Result{Requeue: true}, nil` when we are waiting for something which we know will occur. E.g.: a load balancer external IP provided. 
Although, we should use `ctrl.Result{}, err` when an unplanned error happens.

Reply:
See example: https://github.com/banzaicloud/kafka-operator/blob/d4dfb484f223c0c1062123c0dd6d78e0e86961e9/controllers/kafkacluster_controller.go#L107

Reply:
Makes sense, I'll try to clean it up later